hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonInputFormatUtil.java

```diff
@@ -70,6 +69,18 @@ public static CarbonQueryPlan createQueryPlan(CarbonTable carbonTable, String co
           i++;
         }
       }
+    } else {
+      // query all columns
+      List<CarbonMeasure> tableMsrs = carbonTable.getMeasureByTableName(factTableName);
```

Comment:
yes, But we cannot make it mandatory because in count(*) case it sets null projection so that just returns empty rows but can calculate the count using that.